py/src/braintrust/logger.py

```diff
@@ -1372,6 +1421,58 @@ def __str__(self):
         )
 
 
+class ExperimentDataset:
+    """
+    An `ExperimentDataset` is a dataset that maps to an experiment, and uses its `output` or `expected`
+    fields as the `output` field of the dataset. This is useful for evaluating the performance of your application
+    against a prior experiment, instead of a ground truth value.
+
+    Do not create ExperimentDataset objects directly. Instead, use the `fetch_base_experiment()` method of `Experiment`.
+    """
+
+    def __init__(self, experiment_id: str, output_field: str):
+        self.fetcher = _ObjectLoader("experiment", experiment_id)
+        self.output_field = output_field
+
+    def fetch(self):
+        """
+        Fetch all records in the experiment.
+
+        ```python
+        for record in dataset.fetch():
+            print(record)
+
+        # You can also iterate over the dataset directly.
+        for record in dataset:
+            print(record)
+        ```
+
+        :returns: An iterator over the records in the dataset.
+        """
+        for record in self.fetcher.fetch():
+            if self.output_field == "auto":
+                output = record.get("expected") or record.get("output")
```

Comment:
Don't we need to add an `expected` field to `ObjectLoader.fetch` to make this work?